### PR TITLE
[DOC] Added in new CSAT widget

### LINKED ISSUE #35336: [docs] new CSAT widget design
### Description

The Crisp integration for Kapa (https://github.com/ray-project/ray/pull/35125) will be in the lower right hand corner of the docs pages where the CSAT (thumbs up, thumbs down) widget is currently. We should keep the CSAT because it is giving us valuable signal, but the UI can be improved. @simran has a proposal to move it to the lower left hand side of each doc page with a new design. 

@peytondmurray will also add the GA4 tag so we can see the data in GA4.

### Link

This change would affect every page.

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

This PR removes the existing CSAT widget and adds a new custom one in its place. In addition to reporting upvotes/downvotes, this PR allows both positive and negative written feedback, all through google analytics.

Note that a new GA4 tag will need to be added to `ray/doc/source/_templates/layout.html` in place of the old UA one later on to complete the transition to GA4.

## Related issue number

Closes https://github.com/ray-project/ray/issues/35336.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by simran-2797 — APPROVED** [review]

File: doc/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -a -E
```
Comment (richardliaw) [inline doc/Makefile:5]: hmm whats the reason for removal?

Comment (peytondmurray) [reply doc/Makefile:5]: I added these previously to alleviate some build issues I was having. They force a complete rebuild, which as a general rule is nice, but it takes more time. I can remove this change.

**Review by angelinalg — APPROVED** [review]
Thank you so much @peytondmurray and @simran-2797, for making this so much better!

**Review by matthewdeng — APPROVED** [review]
Test failures look unrelated, merging.

**Pull request merged**